test/extended/prometheus/storage.go

```diff
@@ -0,0 +1,67 @@
+package prometheus
+
+import (
+	"context"
+	"fmt"
+
+	g "github.com/onsi/ginkgo"
+	"github.com/openshift/origin/pkg/test/ginkgo/result"
+	exutil "github.com/openshift/origin/test/extended/util"
+	helper "github.com/openshift/origin/test/extended/util/prometheus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+var _ = g.Describe("[sig-storage][Late] Metrics", func() {
+	defer g.GinkgoRecover()
+	var (
+		oc = exutil.NewCLIWithoutNamespace("prometheus")
+
+		url, bearerToken string
+	)
+
+	g.BeforeEach(func() {
+		var ok bool
+		url, bearerToken, ok = helper.LocatePrometheus(oc)
+		if !ok {
+			e2e.Failf("Prometheus could not be located on this cluster, failing prometheus test")
+		}
+	})
+	g.It("should report short attach times", func() {
+		checkOperation(oc, url, bearerToken, "kube-controller-manager", "volume_attach", 120)
+	})
+	g.It("should report short mount times", func() {
+		checkOperation(oc, url, bearerToken, "kubelet", "volume_mount", 120)
+	})
+})
+
+func checkOperation(oc *exutil.CLI, url string, bearerToken string, component string, name string, threshold int) {
+	plugins := []string{"kubernetes.io/azure-disk", "kubernetes.io/aws-ebs", "kubernetes.io/gce-pd", "kubernetes.io/cinder", "kubernetes.io/vsphere-volume"}
+	oc.SetupProject()
+	ns := oc.Namespace()
+	execPod := exutil.CreateUbiExecPodOrFail(oc.AdminKubeClient(), ns, "execpod", nil)
+	defer func() {
+		oc.AdminKubeClient().CoreV1().Pods(ns).Delete(context.Background(), execPod.Name, *metav1.NewDeleteOptions(1))
+	}()
+
+	tests := map[string]bool{}
+	// Check "[total nr. of ops] - [nr. of ops < threshold] > 0' and expect failure (all ops should be < threshold).
+	// Using sum(max(...)) to sum all kubelets / controller-managers
+	// Adding a comment to make the failure more readable.
+	e2e.Logf("Checking that Operation %s time of plugin %s should be < % seconds", name, plugins, threshold)
+	queryTemplate := `
+# Operation %[4]s time of plugin %[1]s should be < %[2]d seconds
+  sum(max_over_time(storage_operation_duration_seconds_bucket{job="%[3]s",le="+Inf",operation_name="%[4]s",volume_plugin="%[1]s"}[2h]))
```

Comment:
Why do we need to sum the `max_over_time` value?

Reply:
`max_over_time` should return the max. nr of operations in given bucket, i.e. the last value in the time series (let me know if there is a better way to get the last value). `sum` then sums nr. of operations from all kubelets / kube-controller-managers.

Reply:
It sounds right.